packages/libp2p/src/connection-manager/dial-queue.ts

```diff
@@ -230,9 +232,22 @@ export class DialQueue {
         // clean up abort signals/controllers
         signal.clear()
       })
-      .catch(err => {
+      .catch(async err => {
         log.error('dial failed to %s', pendingDial.multiaddrs.map(ma => ma.toString()).join(', '), err)
 
+        if (peerId != null) {
```

Comment:
```suggestion
        if (peerId != null && priority < someVal) {
```

to address manual dials? or do we want a specific dial option to flag whether we will update the LAST_DIAL_FAILURE_KEY for a peer that we only use with auto-dialer?